Acceleratio.XCellKit.Tests/WorkBookTests.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Drawing;
+using System.IO;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+
+namespace Acceleratio.XCellKit.Tests
+{
+    [TestClass]
+    public class WorkBookTests
+    {
+        const string STR_TestOutputPath = "C:\\temp\\test.xlsx";
+        [TestCleanup]
+        public void Cleanup()
+        {
+            if (File.Exists(STR_TestOutputPath))
+            {
+                File.Delete(STR_TestOutputPath);
+            }
+        }
+
+        [TestMethod]
+        public void Save_SingleCell_FileCreated()
+        {
+            var newExcel = new SpreadsheetWorkbook();
+
+            var worksheet = new SpreadsheetWorksheet("Test22");
+            worksheet.AddRow(new SpreadsheetRow() { RowCells = new System.Collections.Generic.List<SpreadsheetCell>() { new SpreadsheetCell() { Value = "test" } } });
+
+            newExcel.AddWorksheet(worksheet);
+            newExcel.Save(STR_TestOutputPath);
+
+            Assert.IsTrue(File.Exists(STR_TestOutputPath));
+        }
+
+        [TestMethod]
+        public void NonStreaming_LargeTable_FileCreated()
+        {
+            var newExcel = new SpreadsheetWorkbook();
+            var columnsCount = 10;
+            var worksheet = new SpreadsheetWorksheet("Test22");
+            for (int i = 0; i < 100000; i++)
+            {
+                var cells = new List<SpreadsheetCell>();
+                for (var j = 0; j < columnsCount; j++)
+                {
+
+                    cells.Add(new SpreadsheetCell()
+                    {
+                        BackgroundColor = Color.Red,
+                        ForegroundColor = Color.Blue,
+                        Font = _font,
+                        Alignment = HorizontalAligment.Center,
+                        Value = $"Ovo je test {i} - {j}"
+                    });
+                }
+
+                var row = new SpreadsheetRow()
+                {
+                    RowCells = cells
+                };
+                worksheet.AddRow(row);
+            }
+
+            newExcel.AddWorksheet(worksheet);
+            newExcel.Save(STR_TestOutputPath);
+
+            Assert.IsTrue(File.Exists(STR_TestOutputPath));
+        }
+
+        [TestMethod]
+        public void Streaming_LargeTable_MemoryConsumptionOk()
```

Comment:
This test shows the most optimal way to export excel?

Reply:
yes, i was thinking also about adding a sample app that would highlight some features.
The test is here more so we can check for regressions in memory and time duration if we change something

Reply:
Sample app would be very nice !